source/slang/slang-ir-autodiff-fwd.cpp

```diff
@@ -1626,6 +1670,8 @@ SlangResult ForwardDiffTranscriber::prepareFuncForForwardDiff(IRFunc* func)
     
     initializeLocalVariables(autoDiffSharedContext->moduleInst->getModule(), func);
 
+    lowerSwizzledStores(autoDiffSharedContext->moduleInst->getModule(), func);
```

Comment:
It will be beneficial to move this to cover all functions and not just the AD ones. We can run the pass again at the end of linkAndOptimizeIR, before hitting emit. A good place to call it might be after `finalizeSpecialization`.

Reply:
A pitfall though is if I have a function taking an `__ref` parameter and `swizzleStore` it, the function will get inlined later, and we probably don't want to lower the swizzled store on that parameter just yet.

Reply:
Hmm.. should we run the pass again after those methods are inlined?

Reply:
That sort of defeats the purpose of the pass, which is removing the concern of `swizzledStore` throughout the codebase...

Reply:
I guess its fine to just run the pass for AD then...

Reply:
It looks like the right choice is to implement a targeted optimization in image load/store lowering pass to detect and combine multiple stores, instead of trying to preserve swizzledStore all over the place.